### PR TITLE
make wifi control work with my Kobo Touch N905

### PR DESCRIPTION
**Pull request by mezzarobba**

### CODE REVIEW

**Comment by poire-z** [comment]
Looks like $INTERFACE is defined and = "eth0" on my GloHD, so that should be a no-change.

Just one thing:
```
# iwconfig | grep ESSID | cut -d\" -f2
lo        no wireless extensions.

BLAHBLAH

# iwconfig >/dev/null
lo        no wireless extensions.
```
so this stderr will probably get logged into crash.log and add some noise there.
You may want to use `os.getenv("INTERFACE")` here too.

File: frontend/device/generic/device.lua [hunk 1]
```diff
@@ -199,7 +199,7 @@ function Device:retrieveNetworkInfo()
     if std_out then
         local result = std_out:read("*all")
         std_out:close()
-        std_out = io.popen('iwconfig eth0 | grep ESSID | cut -d\\" -f2')
+        std_out = io.popen('iwconfig | grep ESSID | cut -d\\" -f2')
```
Comment (Frenzie) [inline frontend/device/generic/device.lua]: Btw, what's the device on the Touch N905? In any case as written this won't work right as a generic statement.

As written:
```sh
# iwconfig | grep ESSID
lo        no wireless extensions.

usb0      no wireless extensions.

eth0      IEEE 802.11-DS  ESSID:"Pink_P"  Nickname:""
```
In other words, you'd get this whole thing:
```sh
# iwconfig | grep ESSID | cut -d\" -f2
lo        no wireless extensions.

usb0      no wireless extensions.

Pink_P
```

Before:
```sh
# iwconfig eth0 | grep ESSID
eth0      IEEE 802.11-DS  ESSID:"Pink_P"  Nickname:""
```

Redirecting stderr to /dev/null works out for me:
```sh
# iwconfig 2>/dev/null | grep ESSID
eth0      IEEE 802.11-DS  ESSID:"Pink_P"  Nickname:""
```

**Comment by mezzarobba** [comment]
Thanks for your comments!

On Tuesday, 1 May 2018 12:11:38 CEST Frans de Jonge wrote:
> Btw, what's the device on the Touch N905? In any case as written this
> won't work right as a generic statement.

While I agree that it is better to redirect stderr to prevent it to get 
logged somewhere, I think it does work as is (provided that there is a 
single wifi interface...). Compare:

```sh
~$ foo=$(iwconfig | grep ESSID | cut -d\" -f2)
lo        no wireless extensions.                                               
                                                                                
eth0      no wireless extensions.                                               
                                                                                
~$ echo $foo
eolienne
```

-- 
Marc

**Comment by mezzarobba** [comment]
On Tuesday, 1 May 2018 12:18:21 CEST Frans de Jonge wrote:
> Incidentally, how do you run KOReader? Also firmware, that kind of
> thing.

This is with the 4.8.10956 firmware, running koreader directly from rcS. 
With this setup, I have to run wpa_supplicant before launching koreader 
for networking control to work.

Essentially, what I'm doing is

```sh

# [stock rcS, minus on-animator, dbus and nickel, then]

mkdir -p /dev/pts
mount -t devpts devpts /dev/pts
/usr/sbin/inetd /etc/inetd.conf

insmod "${WIFI_MODULE_PATH}"
sleep 1
ifconfig ${INTERFACE} up
wpa_supplicant -D wext -s -i ${INTERFACE} -O /var/run/wpa_supplicant -c 
/etc/wpa_supplicant/wpa_supplicant.conf -B

/opt/koreader/koreader.sh


```

-- 
Marc

**Comment by mezzarobba** [comment]
On Tuesday, 1 May 2018 12:11:38 CEST Frans de Jonge wrote:
> Btw, what's the device on the Touch N905?

Do you mean the interface name? wlan0

-- 
Marc

**Comment by Frenzie** [comment]
> I think it does work as is

Right, indeed. It's still noise somewhere. :-)

> This is with the 4.8.10956 firmware, running koreader directly from rcS. With this setup, I have to run wpa_supplicant before launching koreader for networking control to work. Essentially, what I'm doing is

Stock rcS should normally contain this kind of thing:
```sh
PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/lib:
#snip
export PATH PLATFORM PRODUCT CPU runlevel prevlevel
```

> Do you mean the interface name? wlan0

Thanks.

**Comment by mezzarobba** [comment]
On Tuesday, 1 May 2018 12:51:18 CEST Frans de Jonge wrote:
> > This is with the 4.8.10956 firmware, running koreader directly from
> > rcS. With this setup, I have to run wpa_supplicant before launching
> > koreader for networking control to work. Essentially, what I'm
> > doing is
> Stock rcS should normally contain this kind of thing:
> ```sh
> PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/lib:
> #snip
> export PATH PLATFORM PRODUCT CPU runlevel prevlevel
> ```

Hmm, you are right, I don't really need this change. I guess I must have 
added it after launching koreader.sh manually from an ssh session. And I 
think it's still nice to have for that use case, but I'll let you be the 
judge of that.

-- 
Marc Mezzarobba

**Comment by Frenzie** [comment]
Let's keep it but adjust the comment appropriately. `Normally set by rcS, convenience for when launching from SSH` or something along those lines.

**Comment by Frenzie** [comment]
But in that case should we just set the whole $PATH like in rcS?

**Comment by mezzarobba** [comment]
On Tuesday, 1 May 2018 13:11:40 CEST Frans de Jonge wrote:
> But in that case should we just set the whole $PATH like in rcS?

I don't know... so I got around the issue by removing the corresponding 
change from the pull request. It won't be hard to do properly in a 
separate commit if needed.

-- 
Marc Mezzarobba

**Comment by Frenzie** [comment]
Thanks!

**Comment by mezzarobba** [comment]
On Tuesday, 1 May 2018 14:49:49 CEST Frans de Jonge wrote:
> Thanks!

Thank you!

-- 
Marc Mezzarobba

**Pull request merged**